python/gammaloop/__init__.py

```diff
@@ -1,10 +1,86 @@
-from .interface.gammaloop_interface import GammaLoop
-from .misc.common import GL_DEBUG, GL_CONSOLE_HANDLER, register_symbolica, logger, DATA_PATH, GammaLoopError
+import sys
+from subprocess import Popen
+from enum import StrEnum
 
 __version__ = "0.0.1"
 
+GL_PATH = os.path.abspath(os.path.dirname(os.path.realpath(__file__)))
+
+
+class CLIColour(StrEnum):
+    PURPLE = '\033[95m'
+    CYAN = '\033[96m'
+    DARKCYAN = '\033[36m'
+    BLUE = '\033[94m'
+    GREEN = '\033[92m'
+    YELLOW = '\033[93m'
+    RED = '\033[91m'
+    BOLD = '\033[1m'
+    UNDERLINE = '\033[4m'
+    END = '\033[0m'
+
+
+def check_gammaloop_dependencies(clean_dependencies=False, build_dependencies=False, no_gammaloop_python_venv=False):
+
+    gammaloop_root_path = os.path.abspath(GL_PATH)
+
+    if clean_dependencies:
+        print("Cleaning gammaloop dependencies in '%s' using %s./bin/build_dependencies.sh clean%s ..." % (
+            os.path.abspath(os.path.join(gammaloop_root_path, 'dependencies')), CLIColour.GREEN, CLIColour.END))
+        Popen([f"bash {os.path.join('.', 'bin','build_dependencies.sh')} clean",],
+              shell=True, cwd=gammaloop_root_path).wait()
+        print("%sDependencies folder cleaned successfully.%s" %
+              (CLIColour.GREEN, CLIColour.END))
+        if not build_dependencies:
+            sys.exit(0)
+
+    if build_dependencies:
+        print("Building gammaloop dependencies in '%s' using the %s./bin/build_dependencies.sh%s script ..." % (
+            os.path.abspath(os.path.join(gammaloop_root_path, 'dependencies')), CLIColour.GREEN, CLIColour.END))
+        Popen([f"bash {os.path.join('.', 'bin', 'build_dependencies.sh')}",],
+              shell=True, cwd=gammaloop_root_path).wait()
+        if not os.path.isfile(os.path.join(gammaloop_root_path, 'dependencies', 'INSTALLED')):
+            print("%sCould not build the dependencies. Find more information in '%s'.%s" % (
+                CLIColour.RED, os.path.join(gammaloop_root_path, 'dependencies', 'dependency_build.log.log'), CLIColour.END))
+            sys.exit(1)
+        else:
+            print("%sDependencies built successfully.%s" %
+                  (CLIColour.GREEN, CLIColour.END))
+            sys.exit(0)
+
+    if not os.path.isfile(os.path.join(gammaloop_root_path, 'dependencies', 'INSTALLED')):
+        print("\nGammaloop dependencies are %snot installed%s. Run '%sgammaloop --build_dependencies%s' to install them first. Exiting.\n" % (
+            CLIColour.RED, CLIColour.END, CLIColour.GREEN, CLIColour.END))
+        sys.exit(1)
+
+    venv_path = os.path.abspath(os.path.join(
+        gammaloop_root_path, 'dependencies', 'venv'))
+
+    if venv_path != os.path.abspath(os.path.join(os.path.dirname(sys.executable), os.path.pardir)):
+        print("%sWARNING:%s It is recommended to run gammaloop within its Python virtual environment, by issueing the command:\n\n%ssource %s/bin/activate%s\n" % (
```

Comment:
issueing is a misspelling: it should be issuing

Reply:
I actually saw that but since it was not picked up on by my spellchecker I didn't look it up. Should have trusted myself more ^^